tracer/src/Datadog.Trace/Ci/TestModule.cs

```diff
@@ -41,45 +47,115 @@ private TestModule(string name, string? framework, string? frameworkVersion, Dat
         Name = name;
         Framework = framework;
 
-        var tags = new TestModuleSpanTags
+        // if sessionSpanTags is not null then the TestSession is in-proc.
+        // otherwise the TestSession is out of process
+
+        TestModuleSpanTags tags;
+        if (sessionSpanTags is not null)
         {
-            Type = TestTags.TypeTest,
-            Module = name,
-            Framework = framework,
-            FrameworkVersion = frameworkVersion,
-            CIProvider = environment.Provider,
-            CIPipelineId = environment.PipelineId,
-            CIPipelineName = environment.PipelineName,
-            CIPipelineNumber = environment.PipelineNumber,
-            CIPipelineUrl = environment.PipelineUrl,
-            CIJobName = environment.JobName,
-            CIJobUrl = environment.JobUrl,
-            StageName = environment.StageName,
-            CIWorkspacePath = environment.WorkspacePath,
-            GitRepository = environment.Repository,
-            GitCommit = environment.Commit,
-            GitBranch = environment.Branch,
-            GitTag = environment.Tag,
-            GitCommitAuthorDate = environment.AuthorDate?.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture),
-            GitCommitAuthorName = environment.AuthorName,
-            GitCommitAuthorEmail = environment.AuthorEmail,
-            GitCommitCommitterDate = environment.CommitterDate?.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture),
-            GitCommitCommitterName = environment.CommitterName,
-            GitCommitCommitterEmail = environment.CommitterEmail,
-            GitCommitMessage = environment.Message,
-            BuildSourceRoot = environment.SourceRoot,
-            LibraryVersion = TracerConstants.AssemblyVersion,
-            RuntimeName = frameworkDescription.Name,
-            RuntimeVersion = frameworkDescription.ProductVersion,
-            RuntimeArchitecture = frameworkDescription.ProcessArchitecture,
-            OSArchitecture = frameworkDescription.OSArchitecture,
-            OSPlatform = frameworkDescription.OSPlatform,
-            OSVersion = CIVisibility.GetOperatingSystemVersion(),
-        };
-
-        if (environment.VariablesToBypass is { } variablesToBypass)
+            // In-Proc session
+            tags = new TestModuleSpanTags
```

Comment:
Just wondering, is it worth whacking an internal interface on `CIEnvironmentValues` and SessionSpanTags so you can remove the duplication here? Don't know that it is, just an idea...

or... add an internal constructor to TestSessionSpanTags that takes the `CIEnvironmentValues`, as looks like a lot of the setting is duplicated there too?

Reply:
I've changed this in [44a7b7d](https://github.com/DataDog/dd-trace-dotnet/pull/3389/commits/44a7b7d87f83fa43b88233fb52acbb295aad2d15)